### PR TITLE
CORDA-2531 State mappings are now also updated for new reference states.

### PR DESCRIPTION
**Pull request by roger3cev**
Should be straightforward to review. A one-liner change in the vault.

Added a test to prove it works.

### CODE REVIEW

**Comment by GavT** [comment]
@roger3cev Rick is maxed out. Is there another suitable reviewer for this?

**Comment by roger3cev** [comment]
Jose can review!

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by roger3cev** [comment]
@josecoll do you mind reviewing? @rick-r3 is busy with some other stuff. Cheers!

File: node/src/main/kotlin/net/corda/node/services/vault/NodeVaultService.kt [hunk 1]
```diff
@@ -86,7 +86,7 @@ class NodeVaultService(
     override fun start() {
         bootstrapContractStateTypes()
         rawUpdates.subscribe { update ->
-            update.produced.forEach {
+            (update.produced + update.references).forEach {
```
Comment (rick-r3) [inline node/src/main/kotlin/net/corda/node/services/vault/NodeVaultService.kt]: Looks good.  Can you also add a `toSynchronised()` to the end of line 84 where `contractStateTypeMappings` is created please.  Currently it's not thread safe.

Comment (roger3cev) [reply node/src/main/kotlin/net/corda/node/services/vault/NodeVaultService.kt]: Ah yup - cheers! Thanks for the review...!

**Pull request merged**